APIs/JSON/controllers/http/auth.js

```diff
@@ -0,0 +1,124 @@
+    res.writeHeader(
+      "Set-Cookie",
+      `refresh_token=${signedToken}; Max-Age=${isRemove ? 0 : process.env.JWT_REFRESH_TOKEN_EXPIRES_IN}; HttpOnly; SameSite=Lax; Secure;`
+    )
+  }
+
+  #getRefreshTokenCookie(res, req) {
+    const cookieHeader = this.getCookie(req)
+
+    const signedToken = extractRefreshTokenFromCookie(cookieHeader)
+    if (!signedToken) return null
+
+    const unsignedToken = signature.unsign(signedToken.slice(2), process.env.SIGNATURE_SECRET)
+    if (unsignedToken !== false) {
+      return unsignedToken
+    } else {
+      this.#setRefreshTokenCookie(res, signedToken, true)
+      throw new Error(ERROR_STATUES.INCORRECT_TOKEN.message, {
+        cause: {
+          status: ERROR_STATUES.INCORRECT_TOKEN.status,
+          message: ERROR_STATUES.INCORRECT_TOKEN.message,
+        },
+      })
+    }
+  }
+
+  async login(res, req) {
+    try {
+      const refresh_token = this.#getRefreshTokenCookie(res, req)
+
+      const { login, password, access_token, device_id } = await this.parseJsonBody(res)
+      if (!device_id) {
+        throw new Error(ERROR_STATUES.DEVICE_ID_MISSED.message, {
+          cause: { status: ERROR_STATUES.DEVICE_ID_MISSED.status, message: ERROR_STATUES.DEVICE_ID_MISSED.message },
+        })
+      }
+
+      const userAuthOperation = ServiceLocatorContainer.use("UserAuthOperation")
+      const userInfo = { device_id }
+      let needToUpdateRefreshToken = true
+
+      if (login && password) {
+        userInfo.login = login
+        userInfo.password = password
+      } else if (access_token) {
+        userInfo.token = access_token
+        needToUpdateRefreshToken = false
+      } else if (refresh_token) {
+        userInfo.token = refresh_token
+      } else {
+        throw new Error(ERROR_STATUES.MISSING_AUTH_CREDENTIALS.message, {
+          cause: {
+            status: ERROR_STATUES.MISSING_AUTH_CREDENTIALS.status,
+            message: ERROR_STATUES.MISSING_AUTH_CREDENTIALS.message,
+          },
+        })
+      }
+
+      const { user, token: accessToken } = await userAuthOperation.perform(null, userInfo)
+
+      const newRefreshToken = needToUpdateRefreshToken
+        ? await userAuthOperation.createRefreshToken(user, device_id)
+        : null
+
+      const accessTokenExpiredAt =
+        new Date(accessToken.created_at).getTime() + process.env.JWT_ACCESS_TOKEN_EXPIRES_IN * 1000
+
+      if (newRefreshToken) {
+        this.#setRefreshTokenCookie(res, newRefreshToken.token)
+      }
+
+      this.sendSuccess(res, { user, access_token: accessToken.token, expired_at: accessTokenExpiredAt })
+    } catch (err) {
+      console.log(err)
+      this.sendError(res, err.cause?.status || 500, err.cause?.message || ERROR_STATUES.INTERNAL_SERVER.message)
+    }
+  }
+
+  async logout(res, req) {
+    try {
+      const refresh_token = this.#getRefreshTokenCookie(res, req)
+
+      const { device_id } = await this.parseJsonBody(res)
```

Comment:
do not need to pass `device_id`. Only thing we need to pass is `access_token`. Connected device can be retrieved from it.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization